cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -509,7 +506,7 @@ public void upgradeApplication(ApplicationId appId, ProgramTerminator programTer
     Id.Artifact newArtifact = Id.Artifact.fromEntityId(Artifacts.toProtoArtifactId(appId.getParent(), newArtifactId));
     ArtifactDetail newArtifactDetail = artifactRepository.getArtifact(newArtifact);
 
-    updateApplicationInternal(appId, currentSpec.getConfiguration(), programTerminator, newArtifactDetail,
+    updateApplicationInternal(appId, currentSpec.getConfiguration(), programId -> { }, newArtifactDetail,
```

Comment:
Shall we just remove the `terminator` in the `updateApplicationInternal` method? Is it still needed?

Reply:
I kept it so that any "update" usage in future can use it for their purpose. For upgrade, we dont want to terminate jobs but may be for other kind "updates" we might want to so I kept it for now (also it is similar to current "Update" API which updates config in place).